### PR TITLE
Fix 1031: Onboarding Flow landscape

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
## Explanation
Onboarding Flow landscape
## Mocks
https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/1f86ca4e-795e-4a75-8784-fea53864b94b/RG-OB-Onboarding-1-
The following is a list of opportunities to improve the accessibility of Oppia. Each item corresponds to an outlined area on the attached screenshot.

##GIF 
mobizen_20200429_150210 (1)

## Accessibility Scanner Results:
Item label
org.oppia.app:id/onboarding_fragment_next_image_view
This item may not have a label readable by screen readers.

Item descriptions
org.oppia.app:id/slide_image_view
This non-clickable item's speakable text: "Welcome to Oppia!" is identical to that of 1 other item(s).
screenshot_Oppia_2020-04-29-15_08_10

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout-land/onboarding_fragment.xml [hunk 1]
```diff
@@ -31,73 +30,54 @@
...
-        app:layout_constraintBottom_toBottomOf="@+id/slide_dots_container"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintTop_toTopOf="@+id/slide_dots_container" />
-    </androidx.constraintlayout.widget.ConstraintLayout>
+    <ImageView
+      android:id="@+id/onboarding_fragment_next_image_view"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:layout_marginEnd="16dp"
+      android:layout_marginBottom="4dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_fragment.xml]: This is `0dp` in mocks.

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 2]
```diff
@@ -18,20 +17,21 @@
...
-      android:adjustViewBounds="true"
-      android:scaleType="fitXY"
       android:layout_height="0dp"
+      android:adjustViewBounds="true"
       android:contentDescription="@{viewModel.contentDescription}"
+      android:scaleType="fitXY"
       android:src="@{viewModel.slideImage}"
       app:layout_constraintBottom_toBottomOf="parent"
-      app:layout_constraintDimensionRatio="H,4:4.5"
+      app:layout_constraintDimensionRatio="4:4"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml]: The ratio for the image is 10:9
`app:layout_constraintDimensionRatio="H, 10:9"` once this change is done, checkout the circle of smiley face, they look more rounded and works well with mocks.

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 3]
```diff
@@ -18,20 +17,21 @@
...
+      app:layout_constraintStart_toStartOf="parent"
       app:layout_constraintTop_toTopOf="parent" />
 
     <TextView
       android:id="@+id/slide_title_text_view"
       android:layout_width="0dp"
       android:layout_height="wrap_content"
-      android:layout_marginTop="140dp"
+      android:layout_marginTop="80dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml:35]: Add marginStart as `8dp`

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 4]
```diff
@@ -47,9 +47,9 @@
       android:id="@+id/slide_description_text_view"
       android:layout_width="0dp"
       android:layout_height="wrap_content"
-      android:layout_marginTop="8dp"
-      android:layout_marginEnd="40dp"
       android:layout_marginStart="10dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml]: This is mentioned as `18dp` in mocks and the end margin is `38dp` so I guess as you have made `38dp` to `40dp`,  18dp can be made as `20dp`

File: app/src/main/res/layout-land/onboarding_slide_final.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  xmlns:app="http://schemas.android.com/apk/res-auto">
+
+  <data>
+
+    <variable
+      name="viewModel"
+      type="org.oppia.app.onboarding.OnboardingSlideFinalViewModel" />
+  </data>
+
+  <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide_final.xml:12]: All the changes from `onboarding_slide` should be done in this too.

File: app/src/main/res/layout-land/onboarding_slide_final.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      android:textColor="@color/oppiaPrimaryText"
+      android:textSize="16sp"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintHorizontal_bias="0.0"
+      app:layout_constraintStart_toEndOf="@+id/slide_image_view"
+      app:layout_constraintTop_toBottomOf="@+id/slide_title_text_view" />
+
+    <Button
+      android:id="@+id/get_started_button"
+      android:layout_width="240dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide_final.xml]: This should be `match_parent` as you already have margin to take of all the things.

File: app/src/main/res/layout-land/onboarding_slide_final.xml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      android:layout_height="wrap_content"
+      android:layout_marginTop="24dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide_final.xml:68]: This is `28dp` in mocks.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Suggested changes. PTAL

**Comment by rt4914** [comment]
@nikitamarysolomanpvt PTAL non_flaky_tests I think merge with latest develop will fix this.

Also, naming convention for branches does not use `_`, instead it uses `-`, like `onboarding-flow-landscape`. Follow this convention in your other branches. 

Reference: https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change

File: app/src/main/res/layout-land/onboarding_fragment.xml (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_fragment.xml]: Done.

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 2 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_slide.xml]: updated to app:layout_constraintDimensionRatio="H, 10:9"

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_slide.xml:35]: Done

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 4 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_slide.xml]: Added margin start 18dp.

**Comment by veena14cs** [comment]
> @nikitamarysolomanpvt PTAL non_flaky_tests I think merge with latest develop will fix this.
> 
> Also, naming convention for branches does not use `_`, instead it uses `-`, like `onboarding-flow-landscape`. Follow this convention in your other branches.
> 
> Reference: https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change

+1

File: app/src/main/res/layout-land/onboarding_slide_final.xml (hunk 7 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_slide_final.xml:68]: I found them as 24 dp in mock.

File: app/src/main/res/layout-land/onboarding_slide_final.xml (hunk 6 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_slide_final.xml]: Done

File: app/src/main/res/layout-land/onboarding_slide_final.xml (hunk 5 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_slide_final.xml:12]: Done

File: app/src/main/res/layout-land/onboarding_fragment.xml [hunk 8]
```diff
@@ -31,73 +30,54 @@
...
-      android:paddingStart="56dp"
-      android:paddingEnd="28dp"
-      android:paddingTop="206dp"
-      android:visibility="@{viewModel.slideNumber == (viewModel.totalNumberOfSlides - 1) ? View.VISIBLE: View.GONE, default=gone}"
-      app:layout_constraintBottom_toBottomOf="@id/bottom_container"
-      app:layout_constraintEnd_toEndOf="parent"
-      app:layout_constraintTop_toTopOf="@+id/onboarding_slide_view_pager">
+      android:layout_marginStart="20dp"
+      android:layout_marginTop="8dp"
+      android:layout_marginEnd="20dp"
```
Comment (veena14cs) [inline app/src/main/res/layout-land/onboarding_fragment.xml:41]: marginEnd is 28dp in the mock and top margin is 24dp.

File: app/src/main/res/layout-land/onboarding_fragment.xml (hunk 1 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/onboarding_fragment.xml]: +1

**Review by veena14cs — CHANGES_REQUESTED** [review]
Please address the changes.

File: app/src/main/res/layout-land/onboarding_fragment.xml (hunk 8 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout-land/onboarding_fragment.xml:41]: It adjusted as per padding.

**Review by nikitamarysolomanpvt — COMMENTED** [review]
> @nikitamarysolomanpvt PTAL non_flaky_tests I think merge with latest develop will fix this.
> 
> Also, naming convention for branches does not use _, instead it uses -, like onboarding-flow-
>landscape. Follow this convention in your other branches.

In https://github.com/oppia/oppia-android/pull/1033 renamed branch to onboarding-flow-landscape

**Comment by nikitamarysolomanpvt** [comment]
duplicate of #1033

**Pull request closed without merging**